huey/tests/test_storage.py

```diff
@@ -94,3 +97,27 @@ def test_event_iterator(self):
         self.assertEqual(res, 'a')
         res = next(i)
         self.assertEqual(res, 'b')
+
+    def test_conflicting_init_args1(self):
+        self.assertRaises(ValueError, RedisStorage,
+                          url="redis://localhost", host="localhost")
+
+    def test_conflicting_init_args2(self):
+        self.assertRaises(ValueError, RedisStorage,
+                          connection_pool=ConnectionPool(), host="localhost")
+
+    def test_conflicting_init_args3(self):
+        self.assertRaises(ValueError, RedisStorage,
+                          connection_pool=ConnectionPool(), url="redis://localhost")
```

Comment:
You can reduce these into a single test using `itertools.combinations`:

```python
for pair in itertools.combinations((url, connection_pool, host), 2):
   self.assertRaises(ValueError, RedisStorage, ...)
```

That's not exactly what you need, but hopefully that makes sense?